src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -0,0 +1,562 @@
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="Count" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="startIndex" /> is negative or greater than the length of <paramref name="destination" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="destination" /> is <c>null</c>.</exception>
+        static virtual void CopyTo(TSelf vector, T[] destination, int startIndex) => TSelf.CopyTo(vector, destination.AsSpan(startIndex));
+
+        /// <summary>Copies a vector to a given span.</summary>
+        /// <param name="vector">The vector to be copied.</param>
+        /// <param name="destination">The span to which the <paramref name="vector" /> is copied.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static virtual void CopyTo(TSelf vector, Span<T> destination)
+        {
+            if (destination.Length < TSelf.Count)
+            {
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
+            }
+            TSelf.StoreUnsafe(vector, ref MemoryMarshal.GetReference(destination));
+        }
+
+        /// <summary>Creates a new vector with all elements initialized to the specified value.</summary>
+        /// <param name="value">The value that all elements will be initialized to.</param>
+        /// <returns>A new vector with all elements initialized to <paramref name="value" />.</returns>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static abstract TSelf Create(T value);
+
+        /// <summary>Creates a new vector from a given array.</summary>
+        /// <param name="values">The array from which the vector is created.</param>
+        /// <returns>A new vector with its elements set to the first <see cref="Count" /> elements from <paramref name="values" />.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">The length of <paramref name="values" /> is less than <see cref="Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="values" /> is <c>null</c>.</exception>
+        static virtual TSelf Create(T[] values) => TSelf.Create(values.AsSpan());
```

Comment:
Why are all of these virtual rather than abstract?  These are net-new APIs rather than something that each of the VectorXx types already provides?

Reply:
It's the same principle we followed for the generic math interfaces, which is that where a behavior could be trivially provided via a DIM, to do so.

In this case, `Create(T[])` and `Create(ROSpan<T>)` do the same thing with the former forwarding to the latter by default.

Best practice is for someone to always override a DIM (which is something we always do), but it's not strictly required.